src/edu/washington/escience/myria/api/encoding/ParallelDatasetEncoding.java

```diff
@@ -0,0 +1,24 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.api.encoding;
+
+import java.util.Set;
+
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.io.AmazonS3Source;
+
+/**
+ *
+ */
+public class ParallelDatasetEncoding extends MyriaApiEncoding {
+  @Required public RelationKey relationKey;
+  @Required public Schema schema;
+  @Required public AmazonS3Source s3Source;
+  public Character delimiter;
+  public Character escape;
+  public Integer numberOfSkippedLines;
+  public Character quote;
+  public Set<Integer> workers;
```

Comment:
I don't see how the RangePartitionFunction would help. I thought a
RangePartitionFunction meant that we were splitting the data based on
ranges of attribute values. In parallel ingest, we are splitting based on
byte ranges, which seems irrelevant. If we wanted some type of
partitioning, I would think RoundRobinPartitionFunction should still be ok.

On Tue, Jul 26, 2016 at 1:28 PM, Tobin Baker <notifications@github.com>
wrote:

> In
> src/edu/washington/escience/myria/api/encoding/ParallelDatasetEncoding.java
> <https://github.com/uwescience/myria/pull/822#discussion_r72330328>:
>
> > +import edu.washington.escience.myria.RelationKey;
> > +import edu.washington.escience.myria.Schema;
> > +import edu.washington.escience.myria.io.AmazonS3Source;
> > +
> > +/**
> > + *
> > + */
> > +public class ParallelDatasetEncoding extends MyriaApiEncoding {
> > +  @Required public RelationKey relationKey;
> > +  @Required public Schema schema;
> > +  @Required public AmazonS3Source s3Source;
> > +  public Character delimiter;
> > +  public Character escape;
> > +  public Integer numberOfSkippedLines;
> > +  public Character quote;
> > +  public Set<Integer> workers;
>
> DatasetEncoding (which this more or less copies) has a partitionFunction
> field that defaults to RoundRobinPartitionFunction. Should our range
> partitioning be considered equivalent to RoundRobinPartitionFunction for
> the purposes of the optimizer (i.e., it can't make any assumptions about
> partitioning based on tuple values), or should we introduce a new
> RangePartitionFunction (that would presumably be treated the same way by
> the optimizer), to make the distinction between range partitioning and
> round-robin partitioning clear? At least I think we need *some*
> partitioning information in the catalog.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/uwescience/myria/pull/822/files/065f1661b24e07587fe9ca6f887e26ba011c3c34..8dd5e3cfc34d0381f9222654c3bcfd76999f037b#r72330328>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADFus5Qt9jAgHMNZy9YCxmjq2JMdxqb0ks5qZm38gaJpZM4IMdpy>
> .
>